> @lxbsz thanks. I don't quite follow what you mean when you say:
> 
> > Could you add the comments in the commit and add the singed-off-by at the same time ?
> > Thanks

Please see the following commit's format:

```
commit 245914c1446dddf07d5b58b0a7b2060b50fde4d7
Author: Xiubo Li <xiubli@redhat.com>
Date:   Tue Sep 14 16:25:23 2021 +0800

    rbd: switch strtok to strtok_r
    
    The strtok is not thread safe, there could be several threads will
    call the tcmu_rbd_open() at the same time.
    
    And sometimes we can randomly hit some errors like:
    
    [ERROR] tcmu_rbd_open:877 datapool/block0: Could not get image name
    
    But when try it again it will succeed.
    
    Signed-off-by: Xiubo Li <xiubli@redhat.com>
```

--- a/libtcmu.c
+++ b/libtcmu.c
@@ -1109,7 +1109,13 @@ device_cmd_head(struct tcmu_device *dev)
 {
 	struct tcmu_mailbox *mb = dev->map;
 
-	return (struct tcmu_cmd_entry *) ((char *) mb + mb->cmdr_off + mb->cmd_head);
+	/*
+	 * We must load the mb_head index using an atomic load or we'll crash
+	 * on aarch64. See https://github.com/open-iscsi/tcmu-runner/issues/688
+	 */
+	uint32_t mb_head = __atomic_load_n(&mb->cmd_head, __ATOMIC_ACQUIRE);
+	struct tcmu_cmd_entry* e = (struct tcmu_cmd_entry *) ((char *) mb + mb->cmdr_off + mb_head);
+	return e;
 }
 
 static inline struct tcmu_cmd_entry *

